File: backend-model/src/main/java/outcobra/server/model/interfaces/ParentLink.java
package outcobra.server.model.interfaces;

public interface ParentLink {
    Long getId();

    Class<? extends ParentLinked> getParentClass();

    static ParentLink make(Long parentId, Class<? extends ParentLinked> parentClass) {
        return new ParentLink() {
            @Override
            public Long getId() {
                return parentId;
            }

            @Override
            public Class<? extends ParentLinked> getParentClass() {
                return parentClass;
            }
        };
    }
}

File: backend/src/main/kotlin/outcobra/server/model/interfaces/ParentLink.kt
package outcobra.server.model.interfaces

interface ParentLink {
    val id: Long?

    val parentClass: Class<out ParentLinked>

    companion object {

        fun make(parentId: Long?, parentClass: Class<out ParentLinked>): ParentLink {
            return object : ParentLink {
                override val id: Long?
                    get() = parentId

                override val parentClass: Class<out ParentLinked>
                    get() = parentClass
            }
        }
    }
}
